azurerm/internal/services/automation/automation_module_resource.go

```diff
@@ -164,7 +168,13 @@ func resourceArmAutomationModuleCreateUpdate(d *schema.ResourceData, meta interf
 	}
 
 	if _, err := stateConf.WaitForState(); err != nil {
-		return fmt.Errorf("Error waiting for Module %q (Automation Account %q / Resource Group %q) to finish provisioning: %+v", name, accName, resGroup, err)
+		wrappedErr := fmt.Errorf("waiting for Module %q (Automation Account %q / Resource Group %q) to finish provisioning: %+v", name, accName, resGroup, err)
+		if resp, err := client.Delete(ctx, resGroup, accName, name); err != nil {
+			if !utils.ResponseWasNotFound(resp) {
+				wrappedErr = fmt.Errorf("%s\nattempting to delete failed automation account: %+v", wrappedErr, err)
+			}
+		}
+		return wrappedErr
```

Comment:
We shouldn't be randomly deleting things during a creation. This is an API bug, could we report it to them and have them clean up the failed create
```suggestion
		return fmt.Errorf("Error waiting for Module %q (Automation Account %q / Resource Group %q) to finish provisioning: %+v", name, accName, resGroup, err)
```